src/main/java/org/zalando/nakadi/service/timeline/LocalLocking.java

```diff
@@ -0,0 +1,72 @@
+package org.zalando.nakadi.service.timeline;
+
+import java.io.Closeable;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LocalLocking {
+    private static final Logger LOG = LoggerFactory.getLogger(LocalLocking.class);
+    private final Set<String> lockedEventTypes = new HashSet<>();
+    private final Map<String, Integer> eventsBeingPublished = new HashMap<>();
```

Comment:
I did not wanted to suggest a bigger name, but the very fist time I read it I missed event**Type** part of it. Just `event` seems inaccurate. Maybe `eventTypesPublishing` ?